plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -34,6 +36,12 @@ type Query struct {
 	OrderedColumns []string
 }
 
+// var to cache ADAL MSI token
+var adalToken *adal.Token
+
+// var for mutual exclusion lock
+var muCacheLock sync.RWMutex
+
```

Comment:
Is there a reason why you don't move this inside the struct? I assume this is the same for all `servers`!?

Reply:
When I initially wrote this I did have a reason, however now I don't think it makes sense to keep it out of the struct. Moved them to the struct.